### PR TITLE
switch from linktype to dlt everywhere, fixes #1303 and #1479

### LINKED ISSUE #1303: Usage of non-exposed pcap function
**Elasticsearch version**: irrelevant

**Moloch version**: 2.1

**OS name and version**: 5.3.8-1-MANJARO

**How was Moloch built/installed:** (rpm, deb, easybutton, ...): modified install for arch based systems: `./configure --with-libpcap=no --with-yara=no --with-glib2=no --with-pfring=no --with-curl=no --with-lua=no LIBS="-lpcap -lyara -llua -lcurl" GLIB2_CFLAGS="-I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include" GLIB2_LIBS="-lglib-2.0 -lgmodule-2.0 -lgobject-2.0 -lgio-2.0`

**Provide logs, stack traces and steps to reproduce**: (inside easybutton --install, while building capture)
``` 
gcc -rdynamic -ggdb -pthread main.o db.o yara.o http.o config.o parsers.o plugins.o field.o trie.o writers.o writer-inplace.o writer-disk.o writer-null.o writer-simple.o readers.o reader-libpcap-file.o reader-libpcap.o reader-tpacketv3.o reader-null.o packet.o session.o rules.o drophash.o pq.o -o moloch-capture \
            -u g_checksum_update -u g_hmac_update -u g_uri_unescape_segment \
    -lnl-genl-3 -lnl-3 \
    -lglib-2.0 -lgmodule-2.0 -lgobject-2.0 -lgio-2.0 -lmaxminddb -lnet -lrt -luuid -lpcre -lpcap -lyara -llua -lcurl thirdparty/http_parser.o thirdparty/js0n.o thirdparty/patricia.o -ldl -lssl -lcrypto -lyaml \
    -lm -lresolv -lmagic -lffi -lz
/usr/bin/ld: reader-libpcap-file.o: in function `reader_libpcapfile_opened':
/home/m/IdeaProjects/moloch/capture/reader-libpcap-file.c:562: undefined reference to `dlt_to_linktype'
/usr/bin/ld: reader-libpcap.o: in function `reader_libpcap_start':
/home/m/IdeaProjects/moloch/capture/reader-libpcap.c:97: undefined reference to `dlt_to_linktype'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:41: all] Error 1
make[1]: Leaving directory '/home/m/IdeaProjects/moloch/capture'
make: *** [Makefile:329: all-recursive] Error 1
MOLOCH: make failed
```

**Problem:** The used function is only exposed in the (statically linked) .a version of the pcap lib, but not in the .so, as it's not (yet) exposed in the pcap lib, or in any pcap.h. You can test this with `objdump -t libpcap.a | grep dlt` vs `objdump -T libpcap.so | grep dlt`

This PR over there fixes it: https://github.com/the-tcpdump-group/libpcap/pull/882, in the meantime a copy of the fixed libpcap.so is hosted at https://öä.eu/libpcap.so

This issue is a pre-requisite for my efforts to make it arch-linux installable / non-statically-linked installable on systems whose package managers support the according dependencies.

### PR DESCRIPTION
**Pull request by awick**
Switch to using dlt everywhere instead of linktype, when writing a header ourselves do the dlt->linktype conversion with moloch code from information in https://github.com/aol/moloch/issues/1303#issuecomment-554684749

### CODE REVIEW

**Comment by awick** [comment]
@guyharris if you have time to quick review that I implemented your suggestions, that would be awesome

File: capture/packet.c [hunk 1]
```diff
@@ -1207,36 +1208,41 @@ void moloch_packet_batch(MolochPacketBatch_t * batch, MolochPacket_t * const pac
...
-    case 127: // radiotap
+    case DLT_IEEE802_11_RADIO: // radiotap
         rc = moloch_packet_radiotap(batch, packet, packet->pkt, packet->pktlen);
         break;
-    case 228: //RAW IPv4
+    case DLT_IPV4: //RAW IPv4
         rc = moloch_packet_ip4(batch, packet, packet->pkt, packet->pktlen);
         break;
-    case 239: // NFLOG
+    case DLT_IPV6: //RAW IPv4
```
Comment (guyharris) [inline capture/packet.c:1235]: Presumably you mean "RAW IPv6".

**Comment by guyharris** [comment]
Looks good to me.

**Comment by awick** [comment]
thanks!

**Pull request merged**